src/gs-screenshot-image.c

```diff
@@ -368,26 +471,42 @@ gs_screenshot_image_load_async (GsScreenshotImage *ssimg,
 		sizedir = g_strdup_printf ("%ux%u", ssimg->width * ssimg->scale, ssimg->height * ssimg->scale);
 	}
 	cache_kind = g_build_filename ("screenshots", sizedir, NULL);
-	ssimg->filename = gs_utils_get_cache_filename (cache_kind,
-						       basename,
-						       GS_UTILS_CACHE_FLAG_NONE,
-						       NULL);
-	if (ssimg->filename == NULL) {
+
+	/* request the writable cache filename before trying the system's one
+	 * because the former should be the more recent */
+	writable_filename = gs_utils_get_cache_filename (cache_kind,
+							 basename,
+							 GS_UTILS_CACHE_FLAG_WRITEABLE,
+							 NULL);
+
+	if (writable_filename == NULL) {
 		/* TRANSLATORS: this is when we try create the cache directory
 		 * but we were out of space or permission was denied */
 		gs_screenshot_image_set_error (ssimg, _("Could not create cache"));
 		return;
 	}
 
-	/* does local file already exist */
-	if (g_file_test (ssimg->filename, G_FILE_TEST_EXISTS)) {
-		as_screenshot_show_image (ssimg);
-		return;
+	if (ssimg->filename != NULL) {
+		g_free (ssimg->filename);
```

Comment:
But how can `ssimg->filename` not be `NULL` here? The only place we set it is above, but we either return or clear it...